vspec2protobuf.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2021 Motius GmbH
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+#
+# Convert vspec file to proto
+#
+
+import sys
+import vspec
+import getopt
+from anytree import RenderTree, PreOrderIter
+from model.vsstree import VSSNode
+
+mapped = {
+    "uint16": "uint32",
+    "uint8": "uint32",
+    "int8": "int32",
+    "int16": "int32",
+    "boolean": "bool",
+}
+
+
+def traverse_tree(tree, file):
+    tree_node: VSSNode
+    for tree_node in filter(lambda n: n.is_branch(), PreOrderIter(tree)):
+        file.write(f"message {tree_node.qualified_name('')} {{" + "\n")
+        print_message_body(tree_node.children, file)
+        file.write("}\n\n")
+
+
+def print_message_body(nodes, file):
+    for i, node in enumerate(nodes, 1):
+        data_type = node.qualified_name("")
+        if not node.is_branch():
+            dt_val = node.data_type.value
+            data_type = mapped.get(dt_val.strip("[]"), dt_val.strip("[]"))
+            data_type = ("repeated " if dt_val.endswith("[]") else "") + data_type
+        file.write(f"  {data_type} {node.name} = {i};" + "\n")
+
+
+def usage():
+    print(
+        "Usage:",
+        sys.argv[0],
+        "[-I include_dir] ... [-i prefix:id_file] vspec_file output_file",
+    )
+    print("  -I include_dir       Add include directory to search for included vspec")
+    print("                       files. Can be used multiple times.")
+    print()
+    print(
+        "  -i prefix:uuid_file  File to use for storing generated UUIDs for signals with"
```

Comment:
Do we have any coding guidelines for max length of line? The line breaks here makes it difficult to read the code

Reply:
PEP8 suggests a max length of 79 characters. PEP8 is  as far as I know the widest adopted standard and should be the default for teams that are spread throughout different companies. For some reason I could not find strict styling enforcement in this codebase even though it would only take a couple of lines of code to archive that. That is unusual for a codebase like this. Because of that the codebase has a lot of formatting inconsistencies.

I removed the linebreaks.